### PR TITLE
add `always_cast_spell` ammo effect boolean

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Attempt, albeit a bit late, to help with #75079
#### Describe the solution
Add a boolean, that allow spell to be casted if projectile didn't deal any damage, make it false by default
#### Describe alternatives you've considered
i probably could've tried to make a way for ammo_effect to run EoC directly, bypassing the spell part, but i am not familiar with how to make it
#### Testing
GIF 22-07-2024 20-46-07
#### Additional context
@John-Candlebury you added an aftershock gun in #74167 that uses ammo spell, do you want me to add aforementioned boolean to that gun?
Also @Maleclypse

### CODE REVIEW

**Comment by John-Candlebury** [comment]
Both that one and the fusion gun from #74803  should always cast their spells.
Thanks Guardian!

File: data/mods/Aftershock/ammo_effects.json [hunk 1]
```diff
@@ -49,11 +49,13 @@
   {
     "id": "ELECTRIC_CHAIN",
     "type": "ammo_effect",
+    "always_cast_spell ": true,
     "spell_data": { "id": "spell_electric_chain" }
   },
   {
     "id": "FUSION_FAN",
     "type": "ammo_effect",
+    "always_cast_spell ": true,
```
Comment (GuardianDll) [inline data/mods/Aftershock/ammo_effects.json:58]:
```suggestion
    "always_cast_spell": true,
```

File: data/mods/Aftershock/ammo_effects.json [hunk 2]
```diff
@@ -49,11 +49,13 @@
   {
     "id": "ELECTRIC_CHAIN",
     "type": "ammo_effect",
+    "always_cast_spell ": true,
```
Comment (GuardianDll) [inline data/mods/Aftershock/ammo_effects.json:52]:
```suggestion
    "always_cast_spell": true,
```

**Comment by Maleclypse** [comment]
This is good.  I'll make use of this :)

**Pull request merged**